awsproviderlint/passes/AWSAT006/AWSAT006.go

```diff
@@ -0,0 +1,60 @@
+// Package AWSAT006 defines an Analyzer that checks for
+// hardcoded AWS partition DNS suffixes
+package AWSAT006
+
+import (
+	"go/ast"
+	"go/token"
+	"strings"
+
+	"github.com/bflad/tfproviderlint/passes/commentignore"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+const Doc = `check for hardcoded AWS partition DNS suffixes
+
+The AWSAT006 analyzer reports hardcoded AWS partition DNS suffixes. For tests
+to work across AWS partitions, the partition DNS suffixes should not be
+hardcoded.
+`
+
+const analyzerName = "AWSAT006"
+
+var Analyzer = &analysis.Analyzer{
+	Name: analyzerName,
+	Doc:  Doc,
+	Requires: []*analysis.Analyzer{
+		commentignore.Analyzer,
+		inspect.Analyzer,
+	},
+	Run: run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	ignorer := pass.ResultOf[commentignore.Analyzer].(*commentignore.Ignorer)
+	inspect := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+
+	nodeFilter := []ast.Node{
+		(*ast.BasicLit)(nil),
+	}
+	inspect.Preorder(nodeFilter, func(n ast.Node) {
+		x := n.(*ast.BasicLit)
+
+		if ignorer.ShouldIgnore(analyzerName, x) {
+			return
+		}
+
+		if x.Kind != token.STRING {
+			return
+		}
+
+		if !strings.Contains(x.Value, `.amazonaws.com`) {
```

Comment:
Similar to getting the automatically updating list of regions, we can get these DNS suffixes from the `endpoints` package, using the [`(Partition).DNSSuffix` receiver method](https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/#Partition.DNSSuffix). 👍 

```go
resolver := endpoints.DefaultResolver()
partitions := resolver.(endpoints.EnumPartitions).Partitions()
var suffixes []string

for _, p := range partitions {
	suffixes = append(suffixes, p.DNSSuffix())
}
```

Reply:
That's very meta. Don't hardcode the hardcoding test. 🤯